### PR TITLE
[Filter/OpenVino] Add skeleton code of a sub-plugin for OpenVino support

### PR DESCRIPTION
**Pull request by wooksong**
Related to https://github.com/nnsuite/nnstreamer/issues/1708.

This patch adds skeleton code of a tensor filter sub-plugin for OpenVino Toolkit (dldt) support.

Signed-off-by: Wook Song <wook16.song@samsung.com>

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+    for (int k = j; k < NNS_TENSOR_RANK_LIMIT; ++k) {
+      info->info[i].dimension[k] = 1;
+    }
+
+    ieTensorTypeStr = eachOutputInfo->getPrecision ().name ();
+    nnsTensorType = TensorFilterOpenvino::convertFromIETypeStr (ieTensorTypeStr);
+    if (nnsTensorType == _NNS_END) {
+      g_critical ("The type of tensor elements, %s, "
+          "in the model is not supported", ieTensorTypeStr.c_str());
+      goto failed;
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: I think you may need to set `ret` before performing this `goto` statement.

File: ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+    for (int k = j; k < NNS_TENSOR_RANK_LIMIT; ++k) {
+      info->info[i].dimension[k] = 1;
+    }
+
+    ieTensorTypeStr = eachInputInfo->getPrecision().name();
+    nnsTensorType = TensorFilterOpenvino::convertFromIETypeStr(ieTensorTypeStr);
+    if (nnsTensorType == _NNS_END) {
+      g_critical ("The type of tensor elements, %s, "
+          "in the model is not supported", ieTensorTypeStr.c_str());
+      goto failed;
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: Same here

File: ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+  std::size_t ext_start_at = model_path.find_last_of ('.');
+  std::size_t expected_len_path_model_prefix = model_path.length () - 4;
+  std::string path_model_prefix;
+  TensorFilterOpenvino *tfOv;
+
+  if ((ext_start_at == expected_len_path_model_prefix)
+      || (model_path.find (TensorFilterOpenvino::extBin,
+              ext_start_at) != std::string::npos)
+      || (model_path.find (TensorFilterOpenvino::extXml,
+              ext_start_at) != std::string::npos)) {
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: Does this sub-plugin require two model paths (i.e., bin and xml)? Then, how about introducing an additional argument (like caffe2 sug-plugin) without assuming the same prefix? For example, someone may want to use this sub-pluging using different paths (e.g., /etc/A/model.xml, /etc/B/weight.bin).

File: ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+
+/**
+ * @brief Convert the string representing the tensor data type to _nns_tensor_type
+ * @param type a std::string representing the tensor data type in InferenceEngine
+ * @return _nns_tensor_type corresponding to the tensor data type in InferenceEngine if OK, otherwise _NNS_END
+ */
+tensor_type
+TensorFilterOpenvino::convertFromIETypeStr (std::string type)
+{
+  if (type[0] == 'U') {
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: I think it's better to compare the exact string unless `type` has a random suffix, for correct error handling of type conversion.

Comment (dongju-chae) [reply ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: Resolved in offline :)

File: ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc (hunk 2 above)
Comment (wooksong) [reply ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: @dongju-chae Fixed. Thanks.

File: ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc (hunk 1 above)
Comment (wooksong) [reply ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: @dongju-chae Fixed. Thanks.

File: ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc (hunk 3 above)
Comment (wooksong) [reply ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: @dongju-chae Your concern will be applied in another PR.

File: ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+      || (model_path.find (TensorFilterOpenvino::extBin,
+              ext_start_at) != std::string::npos)
+      || (model_path.find (TensorFilterOpenvino::extXml,
+              ext_start_at) != std::string::npos)) {
+    path_model_prefix = model_path.substr (0, expected_len_path_model_prefix);
+  } else {
+    path_model_prefix = model_path;
+  }
+
+  tfOv = new TensorFilterOpenvino (path_model_prefix);
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: plz consider re-open case later.

Comment (myungjoo) [reply ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: is this considered?

Comment (wooksong) [reply ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc]: https://github.com/nnsuite/nnstreamer/pull/1991 handles @jaeyun-jung's comment.

**Pull request merged**